classes/Http/Controller/ProfileController.php

```diff
@@ -147,18 +146,11 @@ public function passwordProcessAction(Request $request)
             'password'  => $request->get('password'),
             'password2' => $request->get('password_confirm'),
         ];
-
-        /** @var \HTMLPurifier $htmlPurifier */
-        $htmlPurifier = $this->service('purifier');
-
-        $form = new SignupForm($formData, $htmlPurifier);
+        $form = new SignupForm($formData, $this->purifier);
         $form->sanitize();
 
-        /** @var Session\Session $session */
-        $session = $this->service('session');
```

Comment:
Since we would otherwise be invoking `$request->getSession()` multiple times, can we maybe just use `$session` here and assign it `$request->getSession()`?

Reply:
See the discussion above.